src/ops/ops.go

```diff
@@ -504,13 +507,134 @@ func (o *ops) RemovePV(pvName string) error {
 	}
+
 	return err
 }
 
+func (o *ops) IsRaidDevice(device string) bool {
+	raidDevices, err := o.getRaidDevices()
+
+	if err != nil {
+		o.log.WithError(err).Errorf("Error occurred while trying to get list of raid devices - continue without cleaning")
+		return false
+	}
+
+	// The device itself or one of its partitions
+	expression, _ := regexp.Compile(device + "[\\d]*")
+
+	for _, raidArrayMembers := range raidDevices {
+		for _, raidMember := range raidArrayMembers {
+			if expression.MatchString(raidMember) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func (o *ops) CleanRaidDevice(device string) error {
+	raidDevices, err := o.getRaidDevices()
+
+	if err != nil {
+		return err
+	}
+
+	for raidDeviceName, raidArrayMembers := range raidDevices {
+		err = o.cleanDeviceFromRaidDevice(device, raidDeviceName, raidArrayMembers)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *ops) getRaidDevices() (map[string][]string, error) {
+	output, err := o.ExecPrivilegeCommand(o.logWriter, "mdadm", "-v", "--query", "--detail", "--scan")
+
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(output, "\n")
+	result := make(map[string][]string)
+
+	/*
+		The output pattern is:
+		ARRAY /dev/md0 level=raid1 num-devices=2 metadata=1.2 name=0 UUID=77e1b6f2:56530ebd:38bd6808:17fd01c4
+		   devices=/dev/vda2,/dev/vda3
+		ARRAY /dev/md1 level=raid1 num-devices=1 metadata=1.2 name=1 UUID=aad7aca9:81db82f3:2f1fedb1:f89ddb43
+		   devices=/dev/vda1
+	*/
+	for i := 0; i < len(lines); {
+		if !strings.Contains(lines[i], "ARRAY") {
+			i++
+			continue
+		}
+
+		fields := strings.Fields(lines[i])
+		raidDeviceName := fields[1]
+		i++
+
+		// Ensuring that we have at least two lines per device.
+		if len(lines) == i {
+			break
+		}
+
+		raidArrayMembersStr := strings.TrimSpace(lines[i])
+		prefix := "devices="
+
+		if !strings.HasPrefix(raidArrayMembersStr, prefix) {
+			continue
+		}
+
+		raidArrayMembersStr = raidArrayMembersStr[len(prefix):]
+		result[raidDeviceName] = strings.Split(raidArrayMembersStr, ",")
+		i++
+	}
+
+	return result, nil
+}
+
+func (o *ops) cleanDeviceFromRaidDevice(deviceName string, raidDeviceName string, raidArrayMembers []string) error {
+	raidStopped := false
+
+	expression, _ := regexp.Compile(deviceName + "[\\d]*")
+
+	for _, raidMember := range raidArrayMembers {
+		// A partition or the device itself is part of the raid array.
+		if expression.MatchString(raidMember) {
+			// Stop the raid device.
+			if !raidStopped {
+				o.log.Info("Stopping raid device: " + raidDeviceName)
+				_, err := o.ExecPrivilegeCommand(o.logWriter, "mdadm", "--stop", raidDeviceName)
+
+				if err != nil {
+					return err
+				}
+
+				raidStopped = true
+			}
+
+			// Clean the raid superblock from the device
+			o.log.Infof("Cleaning raid member %s superblock", raidMember)
+			_, err := o.ExecPrivilegeCommand(o.logWriter, "mdadm", "--zero-superblock", raidMember)
```

Comment:
Does this have the potential to erase partitions on devices that are not the installation device?

Reply:
If yes, we need to bring this up in the UI. We have a warning for all bootable devices being formatted, we'll need to add a warning for devices belonging to a raid array also getting formatted

Reply:
no
if expression.MatchString(raidMember)

Reply:
I mean if `/dev/md0` for example is made up of `/dev/sdb1` and `/dev/sdc1` but the installation device is `/dev/sdb`, will `/dev/sdc1` also have its superblock removed?

Reply:
it will format all the disk participating in the raid. Maybe this is something that should be moved to the discovery stage?
A warning that a picked up boot device is part of a RAID?

Reply:
@omertuc no it didn't only the installation device.
@yevgeny-shnaidman already asked @avishayt and @romfreiman about adding a warning and they said that no warning needed

Reply:
So the answer to "Does this have the potential to delete the superblock on devices that are not the installation device?" is "yes" and "We won't add a warning"?

Because "No" and "we won't add a warning" doesn't make much sense, since if the answer is no, there's no need for a warning?

I'm confused

Reply:
> it will format all the disk participating in the raid. Maybe this is something that should be moved to the discovery stage? A warning that a picked up boot device is part of a RAID?

So that sounds like a "yes".

Why do we need to delete all devices in the RAID? Why not just the device we're interested in installing to?

Reply:
I don't see how this would affect other disks - RAID doesn't work that way.  RAID will replicate changes to the RAID device to each disk, but won't track changes to each disk and sync with each other.

Reply:
Agree with Omer.  If we damage the RAID as a byproduct of cleaning the installation disk, that's fine.  But why go out of our way to clean disks that the user didn't ask us to touch?

Reply:
I mean given

```go
if expression.MatchString(raidMember)
```

It does look like the answer is "no" but looks like this then contradicts @yevgeny-shnaidman 's expectations

Reply:
I don't understand what you mean, @omertuc @avishayt. The other disks are not touched. 
Asked @avishayt and @romfreiman about creating a warning like "Your installation disk is part of a raid array," which isn't related to affecting other disks but to damaging the raid device.

Reply:
As for the raid device itself, I'm not certain if it will be damaged, but even though the user requested us to install on this device

Reply:
i have missed that part of the code where only the installation disk's superblock is removed , @michaellevy101  is right, only the installation disk is reconfigured.

Reply:
Okay so that's a completely different warning, which was not discussed in this thread, you never mentioned you're talking about a different warning. This is the source of the confusion.

Also it seems that only deleting the RAID from the installation device (which is what both me and @avishayt think is the correct thing to do) is NOT what @yevgeny-shnaidman expected, according to his response in this thread, so maybe he has good reasons of why we do need to wipe all devices after all

Reply:
@omertuc see my comment above, code is ok

Reply:
Okay so everyone's aligned, conversation can be resolved